### ISSUE TITLE
Is FedBN supposed to only handle BatchNorm, or also LayerNorm/GroupNorm/etc?

### ISSUE THREAD

**Issue by andreicozma1** [sahiltyagi4/FLORA_beta#70]
From the paper, it feels like it's only supposed to handle BatchNorm, right?

If so, our implementation may not be consistent with the original implementation:

```
==================================================
     Parameter: conv1.weight                   | BN detected: False
     Parameter: conv1.bias                     | BN detected: False
     Parameter: bn1.weight                     | BN detected: True
     Parameter: bn1.bias                       | BN detected: True
     Parameter: conv2.weight                   | BN detected: False
     Parameter: conv2.bias                     | BN detected: False
     Parameter: batch_norm2.weight             | BN detected: True
     Parameter: batch_norm2.bias               | BN detected: True
     Parameter: conv3.weight                   | BN detected: False
     Parameter: conv3.bias                     | BN detected: False
     Parameter: layer_norm.weight              | BN detected: True
     Parameter: layer_norm.bias                | BN detected: True
     Parameter: group_norm.weight              | BN detected: True
     Parameter: group_norm.bias                | BN detected: True
     Parameter: fc.weight                      | BN detected: False
     Parameter: fc.bias                        | BN detected: False
```

Is this expected? @sahiltyagi4

**Comment by andreicozma1** [comment]
I think LayerNorm and GroupNorm would also capture client-specific data statistics that probably shouldn't be aggregated across clients. Let me know your thoughts

**Comment by sahiltyagi4** [comment]
Yeah, you're right. The current logic scans through layer names and flags any param if its name includes 'bn' or 'norm'. So in the current models, layer/group norm are also getting aggregated (note: in the currently evaluated models, there are no batch norm layers so no aggregation would have happened if we strictly match with 'bn' layers). But yeah, FedBN should aggregate only batch-norm layers so it's algorithmically correct to remove 'norm ' in _is_bn_layer() fn and only include 'bn' in the if condition. However, I'm not sure if batch normalization layers would always be initialized with 'bn' in its name.

**Comment by andreicozma1** [comment]
> Yeah, you're right. The current logic scans through layer names and flags any param if its name includes 'bn' or 'norm'. So in the current models, layer/group norm are also getting aggregated (note: in the currently evaluated models, there are no batch norm layers so no aggregation would have happened if we strictly match with 'bn' layers). But yeah, FedBN should aggregate only batch-norm layers so it's algorithmically correct to remove 'norm ' in _is_bn_layer() fn and only include 'bn' in the if condition. However, I'm not sure if batch normalization layers would always be initialized with 'bn' in its name.

Got it, let's keep this issue open for now for a future patch / fix (maybe v1.1.0 milestone) since it's not currently applicable to evaluated models, and could find a robust way to filter only batchnorm-only parameters.
